lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -2597,11 +2605,21 @@ void FIRRTLLowering::initializeRegister(
 
   // Emit the initializer expression for simulation that fills it with random
   // value.
+
   addToIfDefBlock("SYNTHESIS", std::function<void()>(), [&]() {
     addToIfDefBlock(
         "RANDOMIZE_REG_INIT",
         [&]() { regInsertionPoint = builder.saveInsertionPoint(); },
         std::function<void()>());
+
+    addToIfDefBlock(
+        "FIRRTL_BEFORE_INITIAL",
+        [&] {
+          if (!areFIRRTLBeforeAndAfterInitialEmitted)
```

Comment:
Can't these conditionals be moved out of the addToIfDefBlock?

Reply:
No, we can't. `addToIfDefBlock` moves the ifdef op to the current insertion point even if the passed callback is nop. This is necessary to keep the order, "ifdef FIRRTL_BEFORE_INITIAL", "initial" and ""ifdef FIRRTL_AFTER_INITIAL".